arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+    // and treats everything else as normal TeX.
+    var arabic = arabicSymbols.replace(
+      MathJax.Extension.Arabic.arabicLanguageRegExp,
+      '\\fliph{\\text{$1}}'
+    );
+
+    return MathJax.Extension.Arabic.TeX(english, arabic);
+  }
+};
+
+MathJax.Hub.Startup.signal.Post('Arabic TeX Startup');
+
+MathJax.Hub.Register.StartupHook('TeX Jax Ready', function () {
+  MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
+    var TEX = MathJax.InputJax.TeX;
+    var Arabic = MathJax.Extension.Arabic;
+    var texParsePush = TEX.Parse.prototype.Push;
+    var texParseMMLToken = TEX.Parse.prototype.mmlToken;
+    var texParseAlignedArray = TEX.Parse.prototype.AlignedArray;
+    var dict = MathJax.Hub.config.Arabic.dict;
+
+    var englishNumbersRegExp = /[0-9]/g;
+
+    var escapeRegExp = (function () {
+      var reRegExpChar = /[\\^$.*+?()[\]{}|]/g;
+      var reHasRegExpChar = new RegExp(reRegExpChar.source);
+
+      return function (string) {
+        return (string && reHasRegExpChar.test(string))
+          ? string.replace(reRegExpChar, '\\$&')
+          : string;
+      };
+    }());
+
+    var getKeysRegExp = function (map) {
+      var keys = Object.keys(map).sort(function (a, b) {
+        return b.length - a.length;
+      });
+
+      return new RegExp(keys.map(escapeRegExp).join('|'), 'gi');
+    };
+
+    var identifiersMap = MathJax.Hub.config.Arabic.identifiersMap;
+    var identifiersKeysRegExp = getKeysRegExp(identifiersMap);
+
+    var operatorsMap = MathJax.Hub.config.Arabic.operatorsMap;
+    var operatorsKeysRegExp = getKeysRegExp(operatorsMap);
+
+
+    TEX.Definitions.Add({
+      macros: {
+        'ar': 'HandleArabic',
+        'alwaysar': 'MarkAsArabic',
+        'fliph': 'HandleFlipHorizontal',
+        'transx': 'TranslateTeX',
+        'transt': 'TranslateText',
+        'transs': 'TranslateSymbols'
+      }
+    });
+
+
+    TEX.Definitions.Add({
+      macros: function () {
+        var definitions = {};
+
+        Object.keys(dict).forEach(function (key) {
+          var texCommand = dict[key][0];
+          definitions[texCommand] = key;
+        });
+
+        return definitions;
+      }()
+    });
+
+
+    TEX.Parse.Augment(function () {
+      var parsers = {};
+
+      Object.keys(dict).forEach(function (key) {
+        parsers[key] = dict[key][1]; // Parser function
+      });
+
+      return parsers;
+    }());
+
+
+    TEX.Parse.Augment({
+      flipHorizontal: function (token) {
+        token.arabicFlipH = !token.arabicFlipH;
+          // Invert the value, because flipping twice means, it is not flipped
+        return token;
+      },
+      arabicNumber: function (token) {
+        var numbersMap = MathJax.Hub.config.Arabic.numbersMap;
+        var text = token.data[0].data[0];
+        var mapped = text.replace(englishNumbersRegExp, function (m) {
+          return numbersMap[m];
+        });
```

Comment:
I know that this was my suggestion, but there is a further refinement that could be made.  There is no need to create the function each time `arabicNumber()` is called.  So you could use something like

    var repaceNumber = function (m) {return numbersMap[m]};

before line 306 and then

    var mapped = text.replace(englishNumbersRegExp, replaceNumber);

here.  Similarly for the other cases below (lines 331 and 345).